File: python/allergies/allergies.py
ALLERGENS = [
    { 'value': 1, 'name': 'eggs'},
    { 'value': 2, 'name': 'peanuts'},
    { 'value': 4, 'name': 'shellfish'},
    { 'value': 8, 'name': 'strawberries'},
    { 'value': 16, 'name': 'tomatoes'},
    { 'value': 32, 'name': 'chocolate'},
    { 'value': 64, 'name': 'pollen'},
    { 'value': 128, 'name': 'cats'},
]

class Allergies(object):
    def __init__(self, score):
        self._lst = [a['name'] for a in ALLERGENS if score & a['value'] == a['value']]

    def is_allergic_to(self, item):
        return item in self._lst

    @property
    def lst(self):
        return self._lst

File: typescript/allergies/allergies.ts
const ALLERGENS: { [key: number]: string } = {
  1: 'eggs',
  2: 'peanuts',
  4: 'shellfish',
  8: 'strawberries',
  16: 'tomatoes',
  32: 'chocolate',
  64: 'pollen',
  128: 'cats',
}

class Allergies {
  private allergies: string[]
  constructor(score: number) {
    this.allergies = Object.keys(ALLERGENS).reduce((accum: string[], key) => {
      if ((score & Number(key)) === Number(key)) {
        accum.push(ALLERGENS[Number(key)])
      }
      return accum
    }, [])
  }

  allergicTo(allergen: string) {
    return this.allergies.includes(allergen)
  }

  list() {
    return this.allergies
  }
}

export default Allergies